### ISSUE TITLE
The training randomly freezes.

### ISSUE THREAD

**Issue by cabaionut** [NVlabs/neuralangelo#211]
Hello everyone,

I have been trying to solve the following issue for a few days: the training process keeps freezing, and sometimes the entire system becomes unresponsive. It has not been able to surpass 6000 iterations. I have made various adjustments, such as those suggested in the FAQ section (or from other comments), and even tried different versions of CUDA, NVIDIA drivers, and PyTorch, but with no success. Below are the details. If anyone knows of a solution, I would greatly appreciate it.

Thank you!
![Screensh
![Screenshot from 2024-09-30 11-04-51](https://github.com/user-attachments/assets/877fc792-58d4-48a7-9172-b218603d17f1)
ot from 2024-09-30 11-02-15](https://github.com/user-attachments/assets/4ea66fd7-f97f-46c3-9cfe-fb190fea795a)

checkpoint:
    save_epoch: 9999999999
    save_iter: 20000
    save_latest_iter: 9999999999
    save_period: 9999999999
    strict_resume: true
cudnn:
    benchmark: true
    deterministic: false
data:
    name: dummy
    num_images: null
    num_workers: 1
    preload: true
    readjust:
        center:
        - 0.0
        - 0.3
        - 0.0
        scale: 0.75
    root: datasets/bia_ds18
    train:
        batch_size: 1
        image_size:
        - 1897
        - 1066
        subset: null
    type: projects.neuralangelo.data
    use_multi_epoch_loader: true
    val:
        batch_size: 1
        image_size:
        - 533
        - 300
        max_viz_samples: 4
        subset: 1
image_save_iter: 9999999999
inference_args: {}
local_rank: 0
logdir: logs/2024_0930_1010_40_bia
logging_iter: 9999999999999
max_epoch: 9999999999
max_iter: 100000
metrics_epoch: null
metrics_iter: null
model:
    appear_embed:
        dim: 8
        enabled: false
    background:
        enabled: true
        encoding:
            levels: 10
            type: fourier
        encoding_view:
            levels: 3
            type: spherical
        mlp:
            activ: relu
            activ_density: softplus
            activ_density_params: {}
            activ_params: {}
            hidden_dim: 256
            hidden_dim_rgb: 128
            num_layers: 8
            num_layers_rgb: 2
            skip:
            - 4
            skip_rgb: []
        view_dep: true
        white: false
    object:
        rgb:
            encoding_view:
                levels: 3
                type: spherical
            mlp:
                activ: relu_
                activ_params: {}
                hidden_dim: 256
                num_layers: 4
                skip: []
                weight_norm: true
            mode: idr
        s_var:
            anneal_end: 0.1
            init_val: 3.0
        sdf:
            encoding:
                coarse2fine:
                    enabled: true
                    init_active_level: 4
                    step: 5000
                hashgrid:
                    dict_size: 21
                    dim: 4
                    max_logres: 11
                    min_logres: 5
                    range:
                    - -2
                    - 2
                levels: 16
                type: hashgrid
            gradient:
                mode: numerical
                taps: 4
            mlp:
                activ: softplus
                activ_params:
                    beta: 100
                geometric_init: true
                hidden_dim: 256
                inside_out: false
                num_layers: 1
                out_bias: 0.5
                skip: []
                weight_norm: true
    render:
        num_sample_hierarchy: 4
        num_samples:
            background: 32
            coarse: 64
            fine: 16
        rand_rays: 512
        stratified: true
    type: projects.neuralangelo.model
nvtx_profile: false
optim:
    fused_opt: false
    params:
        lr: 0.001
        weight_decay: 0.01
    sched:
        gamma: 10.0
        iteration_mode: true
        step_size: 9999999999
        two_steps:
        - 300000
        - 400000
        type: two_steps_with_warmup
        warm_up_end: 5000
    type: AdamW
pretrained_weight: null
source_filename: projects/neuralangelo/configs/custom/bia.yaml
speed_benchmark: false
test_data:
    name: dummy
    num_workers: 0
    test:
        batch_size: 1
        is_lmdb: false
        roots: null
    type: imaginaire.datasets.images
timeout_period: 9999999
trainer:
    amp_config:
        backoff_factor: 0.5
        enabled: false
        growth_factor: 2.0
        growth_interval: 2000
        init_scale: 65536.0
    ddp_config:
        find_unused_parameters: false
        static_graph: true
    depth_vis_scale: 0.5
    ema_config:
        beta: 0.9999
        enabled: false
        load_ema_checkpoint: false
        start_iteration: 0
    grad_accum_iter: 1
    image_to_tensorboard: false
    init:
        gain: null
        type: none
    loss_weight:
        curvature: 0.0005
        eikonal: 0.1
        render: 1.0
    type: projects.neuralangelo.trainer
validation_iter: 5000
wandb_image_iter: 10000
wandb_scalar_iter: 100

**Comment by cabaionut** [comment]
solved with XMP profile disabled
